index.js

```diff
@@ -79,12 +79,14 @@ module.exports = class ServeDrive extends ReadyResource {
 
     const isHEAD = req.method === 'HEAD'
 
-    if (this._onfilter && !this._onfilter(id, filename)) {
+    if (!(await this._onfilter(id, filename, snapshot))) {
       res.writeHead(404)
       res.end()
       return
     }
 
+    if (this.closing) return
```

Comment:
Double checking the intended flow: this will return without having set the header, so will return a 500 in the end right, due to the final check in _onRequest?

Reply:
its closed, so the connection has be destroy'ed anyway